Getting-Started/Backoffice/Property-Editors/Built-in-Property-Editors/Textbox/index.md

```diff
@@ -43,7 +43,7 @@ Textbox is an HTML input control for text. It can be configured to have a fixed
 ```csharp
 @{
     // Perform an null-check on the field with alias 'pageTitle'
-    if (Model.PageTitle.HasValue())
+    if (Model.PageTitle != null)
```

Comment:
@sofietoft are you sure this is correct?

It's highly likely that PageTitle is a String

and therefore an 'empty string' if somebody didn't fill in a page title if it was non mandatory would mean that the value of Model.PageTitle would be "" and not null

so the check here for null would pass for an empty string, and the markup inside the if statement would be written out..

eg `<p></p>`

But really you don't want that empty markup if there is no PageTitle specified!!!

So you'd either check if (string.IsNullOrWhiteSpace(Model.PageTitle))

or I have a feeling that 'HasValue' or a variation of that is a wrapper for doing that?

Reply:
I'm sire that `Model.PageTitle.HasValue()` doesn't work, and that `Model.PageTitle != null` works when the property has a value.

Would of course be worth testing when the property is empty.

Reply:
I've corrected this in #3180

Reply:
@sofietoft The other syntax that might work here is:

```
if (Model.HasValue(Model.GetModelPropertyType(p=>p.PageTitle).PropertyTypeAlias)){
<p>@Model.PageTitle</p>
}
```
Basically, IPublishedContent/IPublishedElement that Modelsbuilder models inherit from DOES have a HasValue() method to check whether the different type of property eg grid, textstring, date has a value or not - but it requires the alias of the property to do the check...

... so a non-modelsbuilder convert, would be writing

```
if (Model.HasValue("pageTitle")){
<p>@(Model.Value<string>("pageTitle"))</p>
}
```
What I've typed above kind of the strongly-typed version of doing the same with Modelsbuidler to avoid having to type 'pageTitle' - but I'm not sure it's that 'neat' or the 'defined' way: or whether at some point .HasValue() on the property became the 'short hand' for that... but now is lost.

The reason using HasValue is preferred over string.IsNullOrWhiteSpace, is it's a lot more 'friendly', but also then consistent across all the property types...eg not just for strings...

Be interesting to see what's taught on the fundamentals?

Reply:
I'm actually not sure what they're doing in the fundamentals course 🤔 Would be interesting to find out.

I get that used the "hasValue" is better or friendlier. I can update the samples again.
I'll put it on my list.

Thanks for the input! Much appreciated 💪